### DISCUSSION TITLE
pnpm install fails with a core-js-pure postinstall error

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/vercel#9190] [Help]
anyone have a hint what could cause `pnpm install` to fail when deploying to vercel (works fine locally)?

from the logs (with node v18.13.0):

```
.../node_modules/core-js-pure postinstall$ node -e "try{require('./postinstall')}catch(e){}"
.../node_modules/core-js-pure postinstall: node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
.../node_modules/core-js-pure postinstall: node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
.../node_modules/core-js-pure postinstall: Failed
ELIFECYCLE  Command failed with exit code 1.
Error: Command "pnpm install" exited with 1
```

**Comment by stefanprobst** [comment]
this only happens when i have `use-node-version` in `.npmrc`. is this an issue with vercel, or with pnpm?

my `.npmrc`:

```
engine-strict=true
shell-emulator=true
# needs to be commented out, otherwise pnpm install breaks on vercel
# locally, this is not an issue
use-node-version=18.13.0
```

**Reply by amyegan** [reply]
It might be caused by conflicting Node version availability in the build environment. Does it still throw a deployment error if you switch to  [setting Node.js version in your project settings](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version) or defining the node property inside [engines](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines) of a package.json file, like
```
{
  "engines": {
    "node": "18"
  }
}
```
?

**Reply by stefanprobst** [reply]
i have set `"engines": { "node": ">=18" }` in `package.json`, which does lead to this warning about version ranges:

![Screenshot_20230111_080132](https://user-images.githubusercontent.com/20753323/211739862-6ed18d0b-a861-4933-8261-9a7f82488160.png)

and in vercel project settings i have set version `18.x`.

**Reply by stefanprobst** [reply]
changing to `"engines": { "node": "18.x" }` unfortunately still shows a glibc error (although in a different `prepare` step it seems)

```
> run-s setup 'generate:*'

node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
ELIFECYCLE  Command failed with exit code 1.
Error: Command "pnpm install" exited with 1
```

**Reply by dbk91** [reply]
I'm seeing this same thing happen in a postinstall for `pnpm install` command. Not quite sure what to do—we definitely want to get ahead of the Node.js 16 EOL...
